### ISSUE TITLE
Training MTSAC on MetaWorld's MT10 is very unstable

### ISSUE THREAD

**Issue by maciejwolczyk** [rlworkgroup/garage#1561]
Hello! I am trying to reproduce the results for MTSAC for MT10 setting from MetaWorld paper, in particular Figure 8 showing the success rate throughout the training. 

Unfortunately, when running the [example available in this repository](https://github.com/rlworkgroup/garage/blob/master/examples/torch/mtsac_metaworld_mt10.py), I get something completely different - in the first half of training the success rate is progressively growing, outperforming even the results in the paper, but after that the learning deteriorates and the performance becomes close to random:

![image](https://user-images.githubusercontent.com/9852093/84665877-a75e2280-af20-11ea-8cdc-3d771b68f96f.png)

The average return plot is very similar:
![image](https://user-images.githubusercontent.com/9852093/84665925-b5ac3e80-af20-11ea-9dac-bf617c343ae3.png)

Seeds are 1 (orange), 130 (dark blue), 876 (light blue).

I know that the code here is not the same that was used to generate the Figure 8 in the MetaWorld paper (as mentioned in issue #1333), but I am still wondering why the training is so unstable. Do you have a recommendation on how to stabilize the learning process with MTSAC? 

Thanks!

**Comment by krzentner** [comment]
Hi Macie,

We believe a major source of instability in running that example is that reward normalization is being used (see [line 46](https://github.com/rlworkgroup/garage/blob/master/examples/torch/mtsac_metaworld_mt10.py#L46)). This is known to make the Q targets less precise, which we think leads to instability.

However, it's possible this is not the only source of instability. Some instability might be inevitable with the published hyper-parameters. There are also some differences between the exact version of MetaWorld that was used in the paper and the version that was made available online, which we are actively looking into addressing.

Hopefully this is helpful in pointing you in the right direction! Please let me know if this addresses the issue.

**Comment by avnishn** [comment]
Hi @maciejwolczyk,

I'm the author of the implementation that you have used. 

What k.r. has said is true:
1) reward normalization can lead to instabilities when training MTSAC
2) We are looking for a set of hyperparameters that gives rivaling performance to the the current set, but removes the instabilities. 

The reason why we normalize our rewards is because metalworld environments output rewards on different scales. An example of this would be that push-v1 outputs rewards of 1e3 when successful and assemble-v1 outputs rewards of 1e4. 

If we train a policy to learn tasks from these 2 environments, the policy may observe that actions that produce high rewards from assemble should be paid more attention than actions from push. Normalization makes it so that all rewards are treated equally.

**Comment by maciejwolczyk** [comment]
Hi all,

thank you very much for your comments! After disabling the normalization the MT10 results look very similar to those that appeared in the paper. I was playing a bit with finding a different, more stable normalization scheme, but no luck so far. 

I am also trying to replicate the single-task results (i.e. show that SAC can solve each task more or less reliably), which seems to be somewhat problematic. I started with the MT10 hyperparameters and tried changing batch size, buffer size etc. using values from the paper, but even with the best setting after 2M steps only around 6 out of 10 tasks achieve 1.0 success rate. Have you maybe tried that and could share the hyperparameters that work reliably, or some tips to make it work?

Thanks once again!
